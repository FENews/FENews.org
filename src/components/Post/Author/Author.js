import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import { getContactHref } from '../../../utils';
import styles from './Author.module.scss';

export const PureAuthor = ({ data }) => {
  const { author } = data.site.siteMetadata;

  return (
    <div className={styles['author']}>
      <p className={styles['author__bio']}>
        {author.bio}
        <a
          className={styles['author__bio-twitter']}
          href={getContactHref('twitter', author.contacts.twitter)}
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>{author.name}</strong> on Twitter
        </a>
      </p>
    </div>
  );
};

export const Author = (props) => (
  <StaticQuery
    query={graphql`
      query AuthorQuery {
        site {
          siteMetadata {
            author {
              name
              bio
              contacts {       
                twitter
              }
            }
          }
        }
      }
    `}
    render={(data) => <PureAuthor {...props} data={data} />}
  />
);

export default Author;
