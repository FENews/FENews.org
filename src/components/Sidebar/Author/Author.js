import React from 'react';
import { withPrefix, Link } from 'gatsby';
import styles from './Author.module.scss';

const Author = ({ author, isIndex }) => (
  <div className={styles['author']}>
    <Link to="/">
      <img
        src={withPrefix(author.photo)}
        className={styles['author__photo']}
        width="75"
        height="75"
        alt={author.name}
      />
    </Link>

    { isIndex ? (
      <h1 className={styles['author__title']}>
        <Link className={styles['author__title-link']} to="/">{author.name}</Link>
      </h1>
    ) : (
      <h2 className={styles['author__title']}>
        <Link className={styles['author__title-link']} to="/">{author.name}</Link>
      </h2>
    )}
    <p className={styles['author__subtitle']}>{author.bio}</p>
  </div>
);

export default Author;
