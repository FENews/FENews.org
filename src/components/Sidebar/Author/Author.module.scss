@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.author {
  &__photo {
    display: inline-block;
    margin-bottom: 0;
    border-radius: 50%;
    background-clip: padding-box
  }

  &__title {
    font-size: $typographic-base-font-size * 1.125;
    font-weight: 600;
    @include line-height(1.125);
    @include margin(.5, 0, .5, 0);

    &-link {
      color: $color-base;

      &:hover,
      &:focus {
        color: $color-base
      }

    }

  }

  &__subtitle {
    color: $color-gray;
    @include line-height(1);
    @include margin-bottom(1)
  }

}