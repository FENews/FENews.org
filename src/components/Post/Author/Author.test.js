import React from 'react';
import renderer from 'react-test-renderer';
import { PureAuthor as Author } from './Author';

describe('Author', () => {
  it('renders correctly', () => {
    const props = {
      data: {
        site: {
          siteMetadata: {
            author: {
              name: 'test',
              bio: 'test',
              contacts: {
                twitter: 'test'
              }
            }
          }
        }
      }
    };

    const tree = renderer.create(<Author {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
