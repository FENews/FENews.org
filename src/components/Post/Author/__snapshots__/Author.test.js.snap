// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div
  className="author"
>
  <p
    className="author__bio"
  >
    test
    <a
      className="author__bio-twitter"
      href="https://www.twitter.com/test"
      rel="noopener noreferrer"
      target="_blank"
    >
      <strong>
        test
      </strong>
       on Twitter
    </a>
  </p>
</div>
`;
